src/Dapr.Client/Extensions/HttpExtensions.cs

```diff
@@ -0,0 +1,52 @@
+﻿// ------------------------------------------------------------------------
+// Copyright 2024 The Dapr Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------------------------------------------
+
+#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+
+namespace Dapr.Client
+{
+    /// <summary>
+    /// Provides extensions specific to HTTP types.
+    /// </summary>
+    public static class HttpExtensions
+    {
+        /// <summary>
+        /// Appends key/value pairs to the query string on an HttpRequestMessage.
+        /// </summary>
+        /// <param name="message">The message to append the query string parameters to.</param>
+        /// <param name="queryStringParameters">The key/value pairs to populate the query string with.</param>
+        public static void AddQueryParameters(this HttpRequestMessage? message,
```

Comment:
A couple of comments:

 - I feel like `HttpRequestMessage` is too high-level for this; IMO it'd be better to accept/return a `Uri`, which also avoids mutation of the message, which I find a little odd for a method that's not otherwise part of a "builder" pattern.
 - I'd make this class/method `internal` if not made part of `DaprClientGrpc` itself, lest developers start using it as a general-purpose query management utility function and commit us to support the API into the future.

Reply:
Fair on both counts. Updated the extension, the method and the unit tests.